charms/osm/sshproxy.py

```diff
@@ -47,28 +47,31 @@ def __init__(self, hostname: str, username: str, password: str = ""):
         self.password = password
 
     @staticmethod
-    def generate_ssh_key(public=None, private=None):
+    def generate_ssh_key():
         """Generate a 4096-bit rsa keypair."""
         if not os.path.exists(SSHProxy.private_key_path):
-            if public and private:
-                with open(SSHProxy.public_key_path, "w") as f:
-                    f.write(public)
-                    f.close()
-                with open(SSHProxy.private_key_path, "w") as f:
-                    f.write(private)
-                    f.close()
-            else:
-                cmd = "ssh-keygen -t {} -b {} -N '' -f {}".format(
-                    SSHProxy.key_type, SSHProxy.key_bits, SSHProxy.private_key_path,
-                )
+            cmd = "ssh-keygen -t {} -b {} -N '' -f {}".format(
+                SSHProxy.key_type, SSHProxy.key_bits, SSHProxy.private_key_path,
+            )
 
-                try:
-                    check_call(cmd, shell=True)
-                except CalledProcessError:
-                    return False
+            try:
+                check_call(cmd, shell=True)
+            except CalledProcessError:
+                return False
 
         return True
 
+    @staticmethod
+    def write_ssh_keys(public=None, private=None):
+        """Write a 4096-bit rsa keypair."""
+        if public and private:
```

Comment:
Why are you stating public and private as default value None if they always need it to execute the function? Why not just state public and private as required arguments?

Reply:
Good point